### DISCUSSION TITLE
BigNumber percentage with calculation with evm-bn

### DISCUSSION THREAD

**Discussion by decafboba** [PaulRBerg/evm-bn#15] [General]
Hello, I'm trying to write some TypeScript versions of functions using `PRBMath` Solidity Library. One thing I am having trouble with is modeling TypeScript functions the same way as the Solidity ones.

For example, in Solidity you can do something like `.25e18` to model 25% out as a Big Number, and multiple X by `.25e18` to get 25% of X. 
```
uint256 twentyFivePercent = X.mul(.25e18);
```

With TypeScript, I have not been able to find an equivalent, even with these wonderful libraries like `evm-bn`. I believe I'm just doing something wrong here, and would love an example for how I can achieve the above using `evm-bn`.

Thanks!

**Comment by PaulRBerg** [comment]
Hi @kenvogg, `25e18` in Solidity is just `0.25` when working with `evm-bn`.

```typescript
const twentyFivePercent: BigNumber = toBn("0.25");
```

`evm-bn` applies 18 decimals, by default. If you'd like to override that, you can set the 2nd parameter:

```typescript
const twoPointFivePercent: BigNumber = toBn("0.25", 19);
```
